src/tools/simpleClient/main.cpp

```diff
@@ -932,6 +933,45 @@ void handleTorqueMsg(ITorqueControl *torque, const yarp::os::Bottle& cmd,
                         }
                         break;
 
+
+                    case VOCAB_REFG:
+                        {
+                            int n_joint = cmd.get(3).asInt();
+
+                            Bottle& jointList = *(cmd.get(4).asList());
+                            Bottle& refList   = *(cmd.get(5).asList());
+                            int i;
+                            const int joint_size = jointList.size();
+                            const int refs_size  = refList.size();
+
+                            if( (joint_size != n_joint) || (refs_size != n_joint) )
+                            {
+                                yError() << "command malformed, size of joint list and reference list must match the number of joints (" << n_joint << ")";
+                                *rec = false;
+                                break;
+                            }
+
+                            if (n_joint > controlledJoints)
+                            {
+                                yError() << "command malformed, number of joints inserted is bigger than number of available joints (" << controlledJoints << ")";
+                                *rec = false;
+                                break;
+                            }
+
+                                int    *joints = new int[n_joint];       // LATER: optimize to avoid allocation.
```

Comment:
Why later? I guess having a buffer big enough will be sufficient for all `n_joint`, given that the size of the buffer is specified as a specific `n_joint` argument of the setRefTorques call:
~~~
*ok = torque->setRefTorques(n_joint, joints, refs); 
~~~
So even if joints is a vector of size 1000, there are no problems.